### ISSUE TITLE
Use ConfirmChannel in RabbitMQ

### ISSUE THREAD

**Issue by kdi-1** [golevelup/nestjs#96]
Hi.
It would be great and more more reliable if there was a way to use **amqplib.ConfirmChannel** in RabbitMQ.

**Comment by WonderPanda** [comment]
@kd-1 What version are you currently using? Recent versions of the package (I believe it's @golevelup/nest-js rabbitmq 1.09.0 and higher) leverage [`amqp-connection-manager`](https://www.npmjs.com/package/amqp-connection-manager) which exposes a ConfirmChannel

**Comment by kdi-1** [comment]
Thank you for response. 
I assumed so because I saw   [`private _channel?: amqplib.Channel;`](https://github.com/golevelup/nestjs/blob/7b1a665fa7678575fda40dd9ca85ef6d0dd4d580/packages/rabbitmq/src/amqp/connection.ts#L53) in AmqpConnection  class declaration.
I seems that **amqplib.ConfirmChannel** type looks better. 
I see now.

**Comment by MarianaPyrih** [comment]
Hi.
Who can help how better to call waitForConfirms(function(err) {...}) on created channel? 
Public channel of AmqpConnection is defined with amqplib.Channel type, but is ConfirmChannel

**Comment by WonderPanda** [comment]
@MarianaPyrih At the moment I think you should be fine to just cast it using Typescript. I'll look into changing the exposed types in the next release

**Comment by MarianaPyrih** [comment]
@WonderPanda Thank you, that's exactly what I did. I just thought, maybe I needed to go in a different direction to get this functionality, so I decided to ask again.
Also, probably, will be good to add callback as  additional optional argument for `publish` method in connection.ts and pass it to channel.publish, as confirmChannel supports. It will allow to handle case, when each published message is ‘acked’ or ‘nacked’ by the server.
